hr_job_category/models/hr.py

```diff
@@ -42,13 +58,31 @@ def create(self, vals):
         return res
 
     def write(self, vals):
-        if "employee_id" in vals:
-            self.mapped("employee_id").write({"category_ids": [(5,)]})
+        prev_data = self.read(["job_id"])
+        if "employee_id" in vals and self.employee_id != vals.get("employee_id"):
+            self._remove_tags(self.employee_id.id, self.job_id.id)
         res = super().write(vals)
-        if "job_id" in vals or ("employee_id" in vals and vals["employee_id"]):
-            self._tag_employees(vals.get("job_id"))
+        # Go through each record and delete tags associated with the previous
+        # job, then add the tags of the new job.
+        #
+
+        for contract in self:
+            for data in prev_data:
+                if (
+                    data.get("id") == contract.id
+                    and data["job_id"]
+                    and data["job_id"][0] != contract.job_id.id
+                ):
+                    self._remove_tags(contract.employee_id.id, data["job_id"][0])
```

Comment:
Then if prev_data is a dict, it simplifies:
```suggestion
            job_id = prev_data.get(contract.id)
            if job_id:
                contract._remove_tags(job_id)
```